edx_proctoring/templates/proctored_exam/proctoring_launch_callback.html

```diff
@@ -198,13 +198,16 @@
 
       function poll_exam_status() {
         var url = '{{exam_attempt_status_url}}';
-
+        var test_url = '{{exam_submit_url}}';
         $.ajax(url).success(function(data){
           // has the end user completed and submitted the exam in the LMS?!?
-          if (data.status === 'submitted' || data.status === 'error' || data.status === 'verified' || data.status === 'rejected') {
+          if (data.status === 'waiting_for_app_timeout' || data.status === 'error' || data.status === 'verified' || data.status === 'rejected') {
             // Signal that the desktop software should terminate
             // NOTE: This is per the API documentation from SoftwareSecure
             window.external.quitApplication();
+            $.ajax(test_url).success(function(data){
+              // action
+            });
```

Comment:
Have you confirmed that this area of the code will be blocking waiting for the user to acknowledge the modal dialog?